galsim/chromatic.py

```diff
@@ -1252,6 +1189,237 @@ def evaluateAtWavelength(self, wave):
+                          hasattr(flux_ratio,'__call__'))
+        self.separable = obj.separable and not self.chromatic
+
+        if isinstance(obj, InterpolatedChromaticObject) and self.chromatic:
+            import warnings
+            warnings.warn("Cannot render image with chromatic transformation applied to it "
+                            "using interpolation between stored images.  Reverting to "
+                            "non-interpolated version.")
+            obj = obj.original
+
+        if isinstance(obj, InterpolatedChromaticObject):
+            self.original = obj
+            self._jac = jac
+            self._off = offset
+            self._frat = flux_ratio
+            
+        elif isinstance(obj, ChromaticTransformation):
+            self.original = obj.original
+            if hasattr(jac, '__call__'):
+                if hasattr(obj._jac, '__call__'):
+                    self._jac = lambda w: jac(w).dot(obj._jac(w))
+                else:
+                    self._jac = lambda w: jac(w).dot(obj._jac)
+                if hasattr(offset, '__call__'):
+                    if hasattr(obj._off, '__call__'):
+                        self._off = lambda w: jac(w).dot(obj._off(w)) + offset(w)
+                    else:
+                        self._off = lambda w: jac(w).dot(obj._off) + offset(w)
+                else:
+                    if hasattr(obj._off, '__call__'):
+                        self._off = lambda w: jac(w).dot(obj._off(w)) + offset
+                    else:
+                        self._off = lambda w: jac(w).dot(obj._off) + offset
+            else:
+                if hasattr(obj._jac, '__call__'):
+                    self._jac = lambda w: jac.dot(obj._jac(w))
+                else:
+                    self._jac = jac.dot(obj._jac)
+                if hasattr(offset, '__call__'):
+                    if hasattr(obj._off, '__call__'):
+                        self._off = lambda w: jac.dot(obj._off(w)) + offset(w)
+                    else:
+                        self._off = lambda w: jac.dot(obj._off) + offset(w)
+                else:
+                    if hasattr(obj._off, '__call__'):
+                        self._off = lambda w: jac.dot(obj._off(w)) + offset
+                    else:
+                        self._off = jac.dot(obj._off) + offset
+            if hasattr(flux_ratio, '__call__'):
+                if hasattr(obj._frat, '__call__'):
+                    self._frat = lambda w: obj._frat(w) * flux_ratio(w)
+                else:
+                    self._frat = lambda w: obj._frat * flux_ratio(w)
+            else:
+                if hasattr(obj._frat, '__call__'):
+                    self._frat = lambda w: obj._frat(w) * flux_ratio
+                else:
+                    self._frat = obj._frat * flux_ratio
+
+        else:
+            self.original = obj
+            self._jac = jac
+            self._off = offset
+            self._frat = flux_ratio
```

Comment:
Any reason not to call this ```self._flux_ratio```?  While skimming the code I didn't notice this line and only saw ```_frat``` later on, and wondered what it was (seems non-obvious to me).